### PR TITLE
Update linting to use eslint and plugin for typescript

### PR DESCRIPTION
**Pull request by willingc**
- Add config for eslint
- Remove deprecated tslint and config
- Autofix a few errors
- Relax rules from errors to warning so linting passes as before

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -229,14 +236,13 @@
...
     "ts-node": "^8.0.0",
-    "tslint": "^5.12.1",
     "tslint-config-prettier": "^1.17.0",
     "tslint-eslint-rules": "^5.4.0",
     "tslint-react": "^4.0.0",
     "tslint-react-a11y": "^1.0.0",
     "tslint-react-perf": "^0.3.0",
     "typedoc": "^0.17.0",
-    "typescript": "^3.7.3",
+    "typescript": "^3.9.7",
```
Comment (captainsafia) [inline package.json]: Do we need the Typescript upgrade for some reason?

Comment (willingc) [reply package.json]: Nope, I don't think so. Will double check.

Comment (willingc) [reply package.json]: Good catch. Anything over 3.3 is supported.

**Review by captainsafia — APPROVED** [review]
Awesome! Thank you so much, Carol!

**Pull request merged**